doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java

```diff
@@ -700,8 +700,9 @@ public SiteRenderingContext createContextForSkin(
                 if (StringUtils.isNotBlank(toolsPrerequisite)
                         && (current != null)
                         && !matchVersion(current, toolsPrerequisite)) {
-                    throw new RendererException("Cannot use skin: has " + toolsPrerequisite
-                            + " Doxia Sitetools prerequisite, but current is " + current);
+                    throw new RendererException("Cannot use skin as it needs " + toolsPrerequisite
+                            + " Doxia Sitetools, but your configuration uses " + current + "."
+                            + "Check your version setup of plugins and skins to use the same doxia stack.");
```

Comment:
WDYT about "Your current Doxia Sitetools version " + current + " does not match the requirements of the skin (" + toolsPrerequisite + "). In order to fix this choose a skin version which is compatible with this Doxia Sitetools version."

Reply:
Not bad, after all.

Reply:
I ran into this error as my combination of site-plugin-version and skin version did not match. It's not only a matter of the skin, but I will open a new PR with the changes you proposed and we'll see if that gets merged. Thanks.